### ISSUE TITLE
gtk_tree_store_get_path: assertion 'iter->stamp == priv->stamp' failed

### ISSUE THREAD

**Issue by Mno-hime** [lwindolf/liferea#208]
Similar to issue #180: About once a month all items in my "Unread" search folder are marked as read and

```
Gtk-CRITICAL **: gtk_tree_store_get_path: assertion 'iter->stamp == priv->stamp' failed
```

line per item marked as read is to be found in journalctl log. I was always suspecting http://blisty.cz/rss2.rb?all=1 feed since they tend to add and remove items to/from the feed as they are pleased. I added the one-liner patch

```
        if (item) {
+               itemview_select_item (NULL);
                itemlist_remove_item (item);
```

which was supposed to fix the problem to my Fedora 1.10.15 version of liferea and now I was hit by the bug. Now I am running 1.10.16 `liferea --debug-all | tee -a liferea.debug` to have a more verbose log once the bug appears again.

**Comment by lwindolf** [comment]
@Mno-hime Just to make sure I understand the description correctly: 

You believe the bug happens when items are dropped from a feed?

The solution from #180 in 1.10.16 does not help?

**Comment by Mno-hime** [comment]
Ad 1) Correct; that's my guess. Ad 2) It happened for 1.10.15+the patch from #180. Thou technically, I can't say for 1.10.16.

**Comment by Mno-hime** [comment]
Haven't seen it for 2 months. Closing.

**Comment by Mno-hime** [comment]
And now again

```
Gtk-CRITICAL **: gtk_tree_store_get_path: assertion 'iter->stamp == priv->stamp' failed
```

for many times on the log and all feeds were marked as read. Shouldn't this be fixed in liferea-1.10.18?

**Comment by mozbugbox** [comment]
Try to get a backtrace by running liferea with:
```
$ G_DEBUG=fatal-criticals liferea

or

$ liferea --g-fatal-warnings

```

see: [Run GLib Apps](https://developer.gnome.org/glib/stable/glib-running.html#G-DEBUG:CAPS)

**Comment by Mno-hime** [comment]
That ends quickly with:

```
liferea --g-fatal-warnings

(liferea:400): Gtk-CRITICAL **: gtk_event_controller_reset: assertion 'GTK_IS_EVENT_CONTROLLER (controller)' failed
Trace/breakpoint trap (core dumped)
```

**Comment by Mno-hime** [comment]
liferea --debug-all &> liferea.out

```
...
PERF   : = htmlview_render_item took 0,005s
HTML   : writing 19609 bytes to HTML view
GUI    : = itemlist selection took 0,065s
PERF   : = itemlist_selection_changed took 0,065s
DB     : loading item 918
DB     : loading item 918
DB     : loading item 917
GUI    : item list selection changed to "Příběh Bakaly v OKD"
DB     : loading item 917
GUI    : = set read status took 0,001s
HTML   : HTML view: selecting "Příběh Bakaly v OKD"
DB     : loading item 917
DB     : loading item 917
DB     : loading item 917
PERF   : = htmlview_render_item took 0,003s
HTML   : writing 13903 bytes to HTML view
GUI    : = itemlist selection took 0,065s
PERF   : = itemlist_selection_changed took 0,065s
DB     : loading item 917
DB     : loading item 917
DB     : loading item 916
GUI    : item list selection changed to "Plýtvám, plýtváš, plýtváme"
DB     : loading item 916
GUI    : = set read status took 0,001s
HTML   : HTML view: selecting "Plýtvám, plýtváš, plýtváme"
DB     : loading item 916
DB     : loading item 916
DB     : loading item 916
PERF   : = htmlview_render_item took 0,002s
HTML   : writing 12424 bytes to HTML view
GUI    : = itemlist selection took 0,054s
PERF   : = itemlist_selection_changed took 0,054s
DB     : loading item 916
DB     : loading item 916
HTML   : HTML view: removing "Plýtvám, plýtváš, plýtváme"
GUI    : item list selection changed to "Plýtvám, plýtváš, plýtváme"
GUI    : item list selection changed to "(null)"

(liferea:3603): Gtk-CRITICAL **: gtk_tree_store_get_path: assertion 'iter->stamp == priv->stamp' failed
HTML   : HTML view: selecting "Plýtvám, plýtváš, plýtváme"
DB     : loading item 0
DB     : Could not load item with id 0!
DB     : loading item 0
DB     : Could not load item with id 0!
HTML   : writing 972 bytes to HTML view
GUI    : = itemlist selection took 0,054s
PERF   : = itemlist_selection_changed took 0,054s
HTML   : HTML view: removing "Příběh Bakaly v OKD"
GUI    : item list selection changed to "Příběh Bakaly v OKD"
GUI    : item list selection changed to "(null)"

(liferea:3603): Gtk-CRITICAL **: gtk_tree_store_get_path: assertion 'iter->stamp == priv->stamp' failed
HTML   : HTML view: selecting "Příběh Bakaly v OKD"
DB     : loading item 0
DB     : Could not load item with id 0!
DB     : loading item 0
DB     : Could not load item with id 0!
HTML   : writing 972 bytes to HTML view
GUI    : = itemlist selection took 0,055s
PERF   : = itemlist_selection_changed took 0,055s
HTML   : HTML view: removing "Mix měsíce: Genot Centre"
GUI    : item list selection changed to "Mix měsíce: Genot Centre"
GUI    : item list selection changed to "(null)"

(liferea:3603): Gtk-CRITICAL **: gtk_tree_store_get_path: assertion 'iter->stamp == priv->stamp' failed
HTML   : HTML view: selecting "Mix měsíce: Genot Centre"
DB     : loading item 0
DB     : Could not load item with id 0!
DB     : loading item 0
DB     : Could not load item with id 0!
HTML   : writing 972 bytes to HTML view
GUI    : = itemlist selection took 0,055s
PERF   : = itemlist_selection_changed took 0,055s
HTML   : HTML view: removing "Podpora zachování Autonomního sociálního centra Klinika"
GUI    : item list selection changed to "Podpora zachování Autonomního sociálního centra Klinika"
GUI    : item list selection changed to "(null)"
...
```

**Comment by lwindolf** [comment]
The errors with "Could not load item with id 0!" are very suspicious.

**Comment by lwindolf** [comment]
Reading GtkTreeView documentation converting the GtkTreeIter lookup table to GtkTreeRowReference seems like the correct way to fix it.

**Comment by Leiaz** [comment]
This happened again.

```
(liferea:32412): Gtk-CRITICAL **: gtk_tree_store_get_path: assertion 'iter->stamp == priv->stamp' failed

Thread 1 "liferea" received signal SIGTRAP, Trace/breakpoint trap.
0x00007ffff090279b in g_logv () from /usr/lib/libglib-2.0.so.0

#0  0x00007ffff090279b in g_logv () from /usr/lib/libglib-2.0.so.0
No symbol table info available.
#1  0x00007ffff090290f in g_log () from /usr/lib/libglib-2.0.so.0
No symbol table info available.
#2  0x00007ffff2bd4dae in ?? () from /usr/lib/libgtk-3.so.0
No symbol table info available.
#3  0x0000000000460d5f in item_list_view_select (ilv=0xd5dc40, item=0x23875a0) at ../../../src/ui/item_list_view.c:989
        iter = {stamp = -3707296, user_data = 0x42e4e8 <metadata_list_get+35>, user_data2 = 0x46604c, user_data3 = 0x2a1fea0}
        path = 0x0
        treeview = 0xceb000
        selection = 0x122b610
#4  0x000000000044f8bc in itemview_select_item (item=0x23875a0) at ../../../src/ui/itemview.c:231
        ivp = 0xde16e0
#5  0x000000000042cba9 in itemlist_selection_changed (item=0x23875a0) at ../../src/itemlist.c:579
        link = 0x0
        node = 0x12b1ad0
        __PRETTY_FUNCTION__ = "itemlist_selection_changed"
#6  0x000000000045f2c6 in on_itemlist_selection_changed (selection=0x122b610, user_data=0xd5dc40) at ../../../src/ui/item_list_view.c:319
        iter = {stamp = -1411686842, user_data = 0x2020a40, user_data2 = 0x0, user_data3 = 0x0}
        model = 0x1ffc220
        item = 0x23875a0
#7  0x00007ffff0bd1102 in ?? () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#8  0x00007ffff0beb8ed in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#9  0x00007ffff0bec48b in g_signal_emit_by_name () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#10 0x00007ffff2be970d in ?? () from /usr/lib/libgtk-3.so.0
No symbol table info available.
#11 0x00007ffff0bd0edf in g_closure_invoke () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#12 0x00007ffff0be2f92 in ?? () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#13 0x00007ffff0bebbfc in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#14 0x00007ffff0bebfdf in g_signal_emit () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#15 0x00007ffff2bd74f9 in gtk_tree_store_remove () from /usr/lib/libgtk-3.so.0
No symbol table info available.
#16 0x000000000045f5d7 in item_list_view_remove_item (ilv=0xd5dc40, item=0x233dd20) at ../../../src/ui/item_list_view.c:399
        iter = {stamp = -1411686842, user_data = 0x29870c0, user_data2 = 0x0, user_data3 = 0x0}
        __func__ = "item_list_view_remove_item"
#17 0x000000000044f845 in itemview_remove_item (item=0x233dd20) at ../../../src/ui/itemview.c:211
No locals.
#18 0x000000000042bf52 in itemlist_check_for_deferred_action () at ../../src/itemlist.c:190
        id = 134925
        item = 0x233dd20
#19 0x000000000042ca98 in itemlist_selection_changed (item=0x237b830) at ../../src/itemlist.c:558
        __PRETTY_FUNCTION__ = "itemlist_selection_changed"
#20 0x000000000045f2c6 in on_itemlist_selection_changed (selection=0x122b610, user_data=0xd5dc40) at ../../../src/ui/item_list_view.c:319
        iter = {stamp = -1411686842, user_data = 0x29870c0, user_data2 = 0x0, user_data3 = 0x0}
        model = 0x1ffc220
        item = 0x237b830
#21 0x00007ffff0bd1102 in ?? () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#22 0x00007ffff0beb8ed in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#23 0x00007ffff0bec48b in g_signal_emit_by_name () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#24 0x00007ffff2be970d in ?? () from /usr/lib/libgtk-3.so.0
No symbol table info available.
#25 0x00007ffff0bd0edf in g_closure_invoke () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#26 0x00007ffff0be2f92 in ?? () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#27 0x00007ffff0bebbfc in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#28 0x00007ffff0bebfdf in g_signal_emit () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#29 0x00007ffff2bd74f9 in gtk_tree_store_remove () from /usr/lib/libgtk-3.so.0
No symbol table info available.
#30 0x000000000045f5d7 in item_list_view_remove_item (ilv=0xd5dc40, item=0x2388c40) at ../../../src/ui/item_list_view.c:399
        iter = {stamp = -1411686842, user_data = 0x1b0f320, user_data2 = 0x0, user_data3 = 0x0}
        __func__ = "item_list_view_remove_item"
#31 0x000000000044f845 in itemview_remove_item (item=0x2388c40) at ../../../src/ui/itemview.c:211
No locals.
#32 0x000000000042bf52 in itemlist_check_for_deferred_action () at ../../src/itemlist.c:190
        id = 134926
        item = 0x2388c40
#33 0x000000000042ca98 in itemlist_selection_changed (item=0x2386a70) at ../../src/itemlist.c:558
        __PRETTY_FUNCTION__ = "itemlist_selection_changed"
#34 0x000000000045f2c6 in on_itemlist_selection_changed (selection=0x122b610, user_data=0xd5dc40) at ../../../src/ui/item_list_view.c:319
        iter = {stamp = -1411686842, user_data = 0x1b0f320, user_data2 = 0x0, user_data3 = 0x0}
        model = 0x1ffc220
        item = 0x2386a70
#35 0x00007ffff0bd1102 in ?? () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
```
There is A LOT of the selection_changed/check_for_deferred_action/remove_item calls stacked. 

```
#20217 0x00007ffff2bd74f9 in gtk_tree_store_remove () from /usr/lib/libgtk-3.so.0
No symbol table info available.
#20218 0x000000000045f5d7 in item_list_view_remove_item (ilv=0xd5dc40, item=0x1bbab20) at ../../../src/ui/item_list_view.c:399
        iter = {stamp = -1411686842, user_data = 0x293fe40, user_data2 = 0x0, user_data3 = 0x0}
        __func__ = "item_list_view_remove_item"
#20219 0x000000000044f845 in itemview_remove_item (item=0x1bbab20) at ../../../src/ui/itemview.c:211
No locals.
#20220 0x000000000042bf52 in itemlist_check_for_deferred_action () at ../../src/itemlist.c:190
        id = 134937
        item = 0x1bbab20
#20221 0x000000000042ca98 in itemlist_selection_changed (item=0x2986510) at ../../src/itemlist.c:558
        __PRETTY_FUNCTION__ = "itemlist_selection_changed"
#20222 0x000000000045f2c6 in on_itemlist_selection_changed (selection=0x122b610, user_data=0xd5dc40) at ../../../src/ui/item_list_view.c:319
        iter = {stamp = -1411686842, user_data = 0x293fe40, user_data2 = 0x0, user_data3 = 0x0}
        model = 0x1ffc220
        item = 0x2986510
#20223 0x00007ffff0bd11b4 in ?? () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#20224 0x00007ffff0beb8ed in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#20225 0x00007ffff0bec48b in g_signal_emit_by_name () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#20226 0x00007ffff2be970d in ?? () from /usr/lib/libgtk-3.so.0
No symbol table info available.
#20227 0x00007ffff0bd0f85 in g_closure_invoke () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#20228 0x00007ffff0be2f92 in ?? () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#20229 0x00007ffff0bebbfc in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#20230 0x00007ffff0bebfdf in g_signal_emit () from /usr/lib/libgobject-2.0.so.0
No symbol table info available.
#20231 0x00007ffff2bd74f9 in gtk_tree_store_remove () from /usr/lib/libgtk-3.so.0
No symbol table info available.
#20232 0x000000000045f5d7 in item_list_view_remove_item (ilv=0xd5dc40, item=0x2989000) at ../../../src/ui/item_list_view.c:399
        iter = {stamp = -1411686842, user_data = 0x1ba2600, user_data2 = 0x0, user_data3 = 0x0}
        __func__ = "item_list_view_remove_item"
#20233 0x000000000044f845 in itemview_remove_item (item=0x2989000) at ../../../src/ui/itemview.c:211
No locals.
#20234 0x000000000042c927 in itemlist_remove_items (itemSet=0x195d420, items=0x94dba0) at ../../src/itemlist.c:498
        item = 0x2989000
        iter = 0x1ce1580
#20235 0x000000000042b8b6 in itemset_merge_items (itemSet=0x195d420, list=0x94db00, allowUpdates=1, markAsRead=0) at ../../src/itemset.c:426
        iter = 0x0
        droppedItems = 0x94dba0
        items = 0x1a87c60
        i = 0
        max = 100
        length = 64
        toBeDropped = 0
        newCount = 38
        flagCount = 0
        __PRETTY_FUNCTION__ = "itemset_merge_items"
#20236 0x0000000000426108 in feed_process_update_result (subscription=0x12a02b0, result=0x1c37860, flags=0) at ../../src/feed.c:251
        itemSet = 0x195d420
        ctxt = 0x1c04ea0
        node = 0x12a0050
        feed = 0x12a0270
#20237 0x0000000000434922 in subscription_process_update_result (result=0x1c37860, user_data=0x12a02b0, flags=0) at ../../src/subscription.c:233
        subscription = 0x12a02b0
        node = 0x12a0050
        processing = 1
        now = {tv_sec = 1, tv_usec = 29049024}
        __func__ = "subscription_process_update_result"
        __PRETTY_FUNCTION__ = "subscription_process_update_result"
#20238 0x0000000000436927 in update_process_result_idle_cb (user_data=0x19b22c0) at ../../src/update.c:575
        job = 0x19b22c0
#20239 0x00007ffff08fbd1a in g_main_context_dispatch () from /usr/lib/libglib-2.0.so.0
No symbol table info available.
#20240 0x00007ffff08fc0d0 in ?? () from /usr/lib/libglib-2.0.so.0
No symbol table info available.
#20241 0x00007ffff08fc17c in g_main_context_iteration () from /usr/lib/libglib-2.0.so.0
No symbol table info available.
#20242 0x00007ffff10f4d9d in g_application_run () from /usr/lib/libgio-2.0.so.0
No symbol table info available.
#20243 0x0000000000436c05 in main (argc=2, argv=0x7fffffffe4c8) at ../../src/main.c:82
        status = 0
```

It looks like a GtkIter is used uninitialized : 
```
(gdb) br item_list_view_id_to_iter
Breakpoint 1 at 0x45ee7f: file ../../../src/ui/item_list_view.c, line 195.
(gdb) c
Continuing.
[Thread 0x7fff63fff700 (LWP 13654) exited]

Thread 1 "liferea" hit Breakpoint 1, item_list_view_id_to_iter (ilv=0xd5dc40, id=134925, 
    iter=0x7fffffc77830) at ../../../src/ui/item_list_view.c:195
195             if (item_list_view_contains_id (ilv, id)) {
(gdb) n
221             return FALSE;
(gdb) 
222     }
(gdb) 
item_list_view_select (ilv=0xd5dc40, item=0x237b830) at ../../../src/ui/item_list_view.c:987
987                             itemlist_selection_changed (NULL);
(gdb) 
989                     path = gtk_tree_model_get_path (gtk_tree_view_get_model (treeview), &iter);
(gdb) 

(liferea:32412): Gtk-CRITICAL **: gtk_tree_store_get_path: assertion 'iter->stamp == priv->stamp' failed
```

**Comment by lwindolf** [comment]
Should be fixed with #489 to be released with 1.12-rc3
